web-registry/src/hooks/useQueryListClassesWithMetadata.ts

```diff
@@ -0,0 +1,58 @@
+import { useState, useEffect } from 'react';
+
+import { QueryClassesResponse } from '@regen-network/api/lib/generated/regen/ecocredit/v1alpha1/query';
+import { ClassInfo } from '@regen-network/api/lib/generated/regen/ecocredit/v1alpha1/types';
+
+import { getMetadataFromUint8Array } from '../lib/metadata-graph';
+import useQueryListClasses from './useQueryListClasses';
+import { CreditClassMetadataLD } from '../generated/json-ld';
+
+interface ClassInfoWithMetadata extends ClassInfo {
+  metadataJson?: Partial<CreditClassMetadataLD>;
+}
+
+export function useQueryListClassesWithMetadata(): ClassInfoWithMetadata[] {
+  const creditClasses = useQueryListClasses();
+  const [classList, setClassList] = useState<ClassInfoWithMetadata[]>([]);
+
+  useEffect(() => {
+    async function fetchMetadata(
+      creditClasses?: QueryClassesResponse,
+    ): Promise<void> {
+      if (!creditClasses || creditClasses?.classes?.length < 1)
+        return Promise.reject();
+
+      await Promise.all(
+        creditClasses.classes.map(
+          async (creditClass: ClassInfoWithMetadata) => {
+            if (creditClass?.metadata?.length) {
```

Comment:
possibly return early here as well? 

The deep nesting is harder for me to track. That's probably subjective though and isn't as much of an issue in my editor as it is on git, so if you find it easier this way, disregard

Reply:
I prefer returning early as well. Usually I go for it, but sometimes I second-guess it, wondering if it is actually more readable. Less nesting is generally preferable though.